// library/MatchBottom.js
//------------------------------------------------------------------------------
//
//     This code was auto generated.
//
//     Manual changes to this file may cause unexpected behavior in your application.
//     Manual changes to this file will be overwritten if the code is regenerated.
//
//------------------------------------------------------------------------------

const extend = require('js-base/core/extend');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const ImageView = require('sf-core/ui/imageview');
const ImageFillType = require('sf-core/ui/imagefilltype');
const Image = require('sf-core/ui/image');

const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const MatchBottom = extend(FlexLayout)(
  //constructor
  function(_super, props) {
    // initalizes super class for this component scope
    _super(this, props);

    const imgStatisticsStyle = getCombinedStyle(".imageView", {
      width: 40,
      height: 40,
      left: 10,
      imageFillType: ImageFillType.STRETCH,
      image: Image.createFromFile("images://statics.png"),
      top: 20,
      bottom: NaN,
      borderRadius: 5
    });
    var imgStatistics = new ImageView(imgStatisticsStyle);
    this.addChild(imgStatistics);
    this.imgStatistics = imgStatistics;

    const imgCommentsStyle = getCombinedStyle(".imageView", {
      width: 40,
      height: 40,
      left: 30,
      imageFillType: ImageFillType.STRETCH,
      image: Image.createFromFile("images://analysis.png"),
      top: 20,
      bottom: NaN,
      borderRadius: 5
    });
    var imgComments = new ImageView(imgCommentsStyle);
    this.addChild(imgComments);
    this.imgComments = imgComments;

    const imgTamamStyle = getCombinedStyle(".imageView", {
      width: 40,
      height: 40,
      left: NaN,
      imageFillType: ImageFillType.STRETCH,
      image: Image.createFromFile("images://tamam.png"),
      top: 20,
      bottom: NaN,
      right: 10,
      borderRadius: 5,
      alignSelf: FlexLayout.AlignSelf.AUTO,
      positionType: FlexLayout.PositionType.ABSOLUTE
    });
    var imgTamam = new ImageView(imgTamamStyle);
    this.addChild(imgTamam);
    this.imgTamam = imgTamam;

    //assign the children to matchBottom 
    this.children = Object.assign({}, {
      imgStatistics: imgStatistics,
      imgComments: imgComments,
      imgTamam: imgTamam
    });

  });

MatchBottom.defaults = getCombinedStyle(".flexLayout", {
  height: 70,
  width: NaN,
  borderRadius: 10,
  backgroundColor: Color.create(255, 22, 23, 24),
  flexDirection: FlexLayout.FlexDirection.ROW
});

module && (module.exports = MatchBottom);
